contrib/pg_tde/documentation/docs/replication.md

```diff
@@ -1,4 +1,257 @@
+        -- Set this key as the active principal key
+        SELECT pg_tde_set_principal_key('my_primary_principal_key', 'file');
+        ```
+
+2. **Standby:**
+    * Before starting the standby, **securely copy** the entire `pg_tde_file_keys_path` directory and its contents (specifically the `my_primary_principal_key` file) from the primary to the exact same path on the standby (`/var/lib/postgresql/pg_tde_keys`).
+    * Ensure permissions on the standby's directory and key file are identical.
+    * In the standby's `postgresql.conf`, configure `pg_tde_key_provider`, `pg_tde_file_keys_path`, and `pg_tde_principal_key_name` to be **identical** to the primary's configuration.
+    * The standby, upon startup, uses these settings to load the key.
+
+### Scenario 2: HashiCorp Vault Key Provider**
+
+1. **Primary:**
+    * Ensure the principal key (e.g., `my_vault_principal_key`) exists in Vault at the configured path (for example, `secret/data/postgres/keys`).
+    * In `postgresql.conf`:
+
+        ```ini
+        pg_tde_key_provider = 'vault'
+        pg_tde_vault_url = 'http://your-vault-server:8200'
+        pg_tde_vault_token = 'your-vault-access-token' # Or use other auth methods
+        pg_tde_vault_mount_path = 'secret/' # Path where KV engine is mounted
+        # pg_tde_vault_ca_path = '/path/to/vault_ca.crt' # If using self-signed certs
+        pg_tde_principal_key_name = 'my_vault_principal_key' # Key name within Vault path
+        ```
+
+    * After restarting PostgreSQL, in `psql`:
+
+        ```sql
+        CREATE EXTENSION IF NOT EXISTS pg_tde;
+        -- Add the key definition (pg_tde will fetch it from Vault)
+        SELECT pg_tde_principal_key_add('my_vault_principal_key', NULL, 'vault');
+        -- Set this key as the active principal key
+        SELECT pg_tde_set_principal_key('my_vault_principal_key', 'vault');
+        ```
+
+2. **Standby:**
+    * In the standby's `postgresql.conf`, configure `pg_tde_key_provider` and all `pg_tde_vault_*` parameters, and `pg_tde_principal_key_name` to be **identical** to the primary's configuration.
+    * Ensure the standby server has network access to the Vault server and the necessary credentials/permissions to read the *same principal key* from Vault.
+    * The standby, upon startup, uses these settings to load the key.
+
+**Is it possible to use the same key provider and principal key?**
+
+**Yes, it is mandatory.** For physical replication to function with TDE, the standby *must* use the same principal key (by name and content, accessed via the same provider configuration) as the primary to decrypt the replicated data blocks.
+
+## Encrypted Table Replication
+
+**If a table on the primary site is encrypted, will all the standby sites also have the encrypted table replicated, given that the same key is implemented on the standby sites as well?**
+
+Yes.
+
+Physical replication works at the block level.
```

Comment:
Should this be a heading? Also, the description in this section is completely incorrect.